extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/KubernetesTaskRunnerFactory.java

```diff
@@ -84,18 +93,58 @@ public KubernetesTaskRunner build()
       client = new DruidKubernetesClient();
     }
 
-    K8sTaskAdapter adapter;
-    if (kubernetesTaskRunnerConfig.sidecarSupport) {
-      adapter = new MultiContainerTaskAdapter(
+    runner = new KubernetesTaskRunner(
+        buildTaskAdapter(client),
+        kubernetesTaskRunnerConfig,
+        taskQueueConfig,
+        taskLogPusher,
+        new DruidKubernetesPeonClient(client, kubernetesTaskRunnerConfig.namespace, kubernetesTaskRunnerConfig.debugJobs)
+    );
+    return runner;
+  }
+
+  @Override
+  public KubernetesTaskRunner get()
+  {
+    return runner;
+  }
+
+  private TaskAdapter buildTaskAdapter(DruidKubernetesClient client)
+  {
+    String adapter = properties.getProperty(String.format(
+        Locale.ROOT,
+        "%s.%s.adapter.type",
+        IndexingServiceModuleHelper.INDEXER_RUNNER_PROPERTY_PREFIX,
+        TYPE_NAME
+    ));
+
+    if (adapter != null && !MultiContainerTaskAdapter.TYPE.equals(adapter) && kubernetesTaskRunnerConfig.sidecarSupport) {
+      throw new IAE(
+          "Only kubernetes pod adapter [%s] can be specified when sidecarSupport is enabled",
```

Comment:
Please also add the value of `adapter` in this message.

Reply:
The value of the adapter is interpolated here, check the next line that specifies `MultiContainerTaskAdapter.TYPE` as a format argument.

Reply:
yes. that's the adapter that user has to specify. And then there is the adapter that user has currently specified. we should log both.